File: lista-de-exercicios/exercicio-6/Aleatorizador.test.ts
import { jest } from '@jest/globals';
import { Aleatorizador } from './Aleatorizador';

const aleatorizador = new Aleatorizador();

describe('Classe Aleatorizador', () => {
    test('Deve conter um array de 10 números', () => {
        expect(aleatorizador.getAleatorios()).toHaveLength(10);
    })

    test('Deve retorar o menor e o maior númerno de um array', () => {
        expect(aleatorizador.getMaiorMenor([1,2,3])).toBe('Menor: 1\nMaior: 3');
    })
})

File: lista-de-exercicios-java/src/test/java/org/example/exercicio06/AleatorizadorTest.java
package org.example.exercicio06;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class AleatorizadorTest {

    Aleatorizador aleatorizador = new Aleatorizador();
    @DisplayName("Deve gerar um array de 10 números")
    @Test
    void deveGerarArrayCom10Numeros() {
        int[] aleatorios = aleatorizador.getAleatorios(10);

        assertEquals(10, aleatorios.length);
    }

    @DisplayName("Deve dar erro se o número for negativo ou zero")
    @Test
    void deveDarErroSeNumeroForNegativoOuZero() {
        assertThrows(IllegalArgumentException.class, () -> {
            aleatorizador.getAleatorios(-5);
        });
        assertThrows(IllegalArgumentException.class, () -> {
            aleatorizador.getAleatorios(0);
        });
    }

    @DisplayName("Deve retornar o maior e o menor número do array")
    @Test
    void deveRetornarMaiorEMenorNumeroDoArray() {
        int[] numeros = {10, 50, 20, 30, 5};

        assertEquals("Menor: 5\nMaior: 50", aleatorizador.getMaiorMenor(numeros));
    }

    @DisplayName("Deve dar erro se o array for vazio")
    @Test
    void deveDarErroSeArrayForVazio() {
        int[] numeros = {};

        assertThrows(IllegalArgumentException.class, () -> {
            aleatorizador.getMaiorMenor(numeros);
        });
    }
}